File: ccif/src/java/other/org/himalay/ccif/MAC_addressReport.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:57 EST 2016")

public  class MAC_addressReport extends BinStruct implements PublicBinMsg{




    // members variables
    // header
    public APDUHeader header ;
    // numberOfAddress
    public ByteArray numberOfAddress ;

        public MAC_addressReport () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize header
        header = new APDUHeader();
        // Initialize numberOfAddress
        numberOfAddress= new ByteArray();numberOfAddress.setSizeType("FIRST_UI8");
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read header
        retVal += header.read(istream);
        // read numberOfAddress
        {retVal+=numberOfAddress.read(istream); }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        { /** fix dependent sizes for header **/  }
        { /** fix dependent sizes for numberOfAddress**/  }
    
                // write header
        if (header!=null)retVal +=header.write(ostream);
        // write numberOfAddress
        {retVal += numberOfAddress.write(ostream);}
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("MAC_addressReport\n");
    dc.increaseIndent();
        int retVal= 0;
        // write header
        if ( header != null ) {dc.indent();dc.getPs().println("header") ;retVal +=header.dump(dc);}
        // write numberOfAddress
        dc.indent();dc.getPs().print("numberOfAddress: "+numberOfAddress.getSize()+"(0x"+Integer.toHexString(numberOfAddress.getSize())+")\n");this.numberOfAddress.dump(dc);
dc.decreaseIndent();
        return retVal;
    }


        // Getter for header
    //public APDUHeader getHeader()
    //{
    //    return header ;
    //}

    
    // Setter for header
    //public void setHeader(APDUHeader val)
    //{
    //    this.header= val;
    //}
    // Getter for numberOfAddress
    //public ByteArray getNumberOfAddress()
    //{
    //    return numberOfAddress ;
    //}

    
    // Setter for numberOfAddress
    //public void setNumberOfAddress(ByteArray val)
    //{
    //    this.numberOfAddress= val;
    //}


    public void setNumberOfAddress(byte[] val)
    {
        this.numberOfAddress.setData(val);
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/MAC_addressReport.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:15 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "MAC_addressReport.h"

#ifdef org_himalay_ccif_MAC_addressReport__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
MAC_addressReport::MAC_addressReport() {
    references= 0;


    // header
    org_himalay_ccif_APDUHeader__NEW( header);
    // numberOfAddress
    org_himalay_msgs_runtime_ByteArray_NEW(numberOfAddress);
    numberOfAddress->setSizeType("FIRST_UI8");
}
// Destructor
MAC_addressReport::~MAC_addressReport() {
    // header
    org_himalay_ccif_APDUHeader__DELETE( header);
    // numberOfAddress
    org_himalay_msgs_runtime_ByteArray_DELETE( numberOfAddress);
}

int MAC_addressReport::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // header
    retVal += header->read(istream);
    // numberOfAddress
    {
        retVal+=numberOfAddress->read(istream);
    }
    return retVal;
}

int MAC_addressReport::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read header
    retVal += header->read(istream);
    // read numberOfAddress
    {
        retVal+=numberOfAddress->read(istream);
    }

    return retVal;
}

int MAC_addressReport::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for header **/
    }
    {   /** fix dependent sizes for numberOfAddress**/
    }

    int retVal= 0;
    // write header
    if ( header != NULL ) {
        retVal +=header->write(ostream);
    }
    // write numberOfAddress
    {
        retVal += numberOfAddress->write(ostream);
    }

    return retVal;
}

int MAC_addressReport::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("MAC_addressReport\n");
    dc.increaseIndent();
    int retVal= 0;
    // write header
    if ( header != NULL ) {
        dc.indent();
        dc.getPs().println("header") ;
        retVal +=header->dump(dc);
    }
    // write numberOfAddress
    dc.indent();
    dc.getPs().print("numberOfAddress");
    numberOfAddress->dump(dc);
    dc.decreaseIndent();
    return retVal;
}


// Getter for header
//org_himalay_ccif_APDUHeader__PTR_TYPE MAC_addressReport::getHeader()
//{
//return this->header;
//}

// Setter for header
//void MAC_addressReport::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE val)
//{
//this->header= val;
//}

// Getter for numberOfAddress
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE MAC_addressReport::getNumberOfAddress()
//{
//return this->numberOfAddress;
//}

// Setter for numberOfAddress
//void MAC_addressReport::setNumberOfAddress(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->numberOfAddress= val;
//}

int MAC_addressReport::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_MAC_addressReport__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::MAC_addressReport* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::MAC_addressReport* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code